pkg/kubectl/cmd/util/factory.go

```diff
@@ -940,16 +940,44 @@ func GetFirstPod(client coreclient.PodsGetter, namespace string, selector labels
 	return pod, 1, nil
 }
 
-// TODO: We need to filter out stuff like secrets.
-func (f *factory) Command() string {
+func (f *factory) Command(enableSecret bool) string {
 	if len(os.Args) == 0 {
 		return ""
 	}
 	base := filepath.Base(os.Args[0])
 	args := append([]string{base}, os.Args[1:]...)
+
+	for i := 0; i < len(args); i++ {
+		if strings.HasPrefix(args[i], "--") {
```

Comment:
This does not handle 'short' flags. Which stinks. Should we find a way to library-ify the command line parsing logic in pflag? `spf13/pflag/flags.go::parseArgs()` is quite complex to get right...